services/motion/builtin/builtin_test.go

```diff
@@ -1731,25 +1279,897 @@ func TestMoveOnGlobe(t *testing.T) {
-		t,
-		"slam/example_cartographer_outputs/viam-office-02-22-3/pointcloud/pointcloud_4.pcd",
-		110,
-		nil,
-	)
-	defer ms.Close(ctx)
+	t.Run("Long distance", func(t *testing.T) {
+		if runtime.GOARCH == "arm" {
+			t.Skip("skipping on 32-bit ARM, large maps use too much memory")
+		}
+		extra := map[string]interface{}{"smooth_iter": 0, "motion_profile": "position_only"}
+		// goal position is scaled to be in mm
+		goalInBaseFrame := spatialmath.NewPoseFromPoint(r3.Vector{X: -32.508 * 1000, Y: -2.092 * 1000})
+		goalInSLAMFrame := spatialmath.PoseBetweenInverse(motion.SLAMOrientationAdjustment, goalInBaseFrame)
 
-	req := motion.MoveOnMapReq{
-		ComponentName: base.Name(),
-		Destination:   spatialmath.NewZeroPose(),
-		SlamName:      slam.Named("test_slam"),
-	}
+		kb, ms := createMoveOnMapEnvironment(
+			ctx,
+			t,
+			"slam/example_cartographer_outputs/viam-office-02-22-3/pointcloud/pointcloud_4.pcd",
+			110,
+			spatialmath.NewPoseFromPoint(r3.Vector{0, -1600, 0}),
+		)
+		defer ms.Close(ctx)
+		req := motion.MoveOnMapReq{
+			ComponentName: base.Named("test-base"),
+			Destination:   goalInSLAMFrame,
+			SlamName:      slam.Named("test_slam"),
+			Extra:         extra,
+		}
+		executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, executionID, test.ShouldNotResemble, uuid.Nil)
+
+		timeoutCtx, timeoutFn := context.WithTimeout(ctx, time.Second*5)
+		defer timeoutFn()
+		err = motion.PollHistoryUntilSuccessOrError(timeoutCtx, ms, time.Millisecond*5, motion.PlanHistoryReq{
+			ComponentName: req.ComponentName,
+			ExecutionID:   executionID,
+			LastPlanOnly:  true,
+		})
+		test.That(t, err, test.ShouldBeNil)
+
+		endPos, err := kb.CurrentPosition(ctx)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, spatialmath.PoseAlmostCoincidentEps(endPos.Pose(), goalInBaseFrame, 15), test.ShouldBeTrue)
+	})
+
+	t.Run("Plans", func(t *testing.T) {
+		// goal x-position of 1.32m is scaled to be in mm
+		// Orientation theta should be at least 3 degrees away from an integer multiple of 22.5 to ensure the position-only test functions.
+		goalInBaseFrame := spatialmath.NewPose(r3.Vector{X: 1.32 * 1000, Y: 0}, &spatialmath.OrientationVectorDegrees{OZ: 1, Theta: 180})
+		goalInSLAMFrame := spatialmath.PoseBetweenInverse(motion.SLAMOrientationAdjustment, goalInBaseFrame)
+		extra := map[string]interface{}{"smooth_iter": 0}
+		extraPosOnly := map[string]interface{}{"smooth_iter": 5, "motion_profile": "position_only"}
+
+		t.Run("ensure success of movement around obstacle", func(t *testing.T) {
+			kb, ms := createMoveOnMapEnvironment(ctx, t, "pointcloud/octagonspace.pcd", 40, nil)
+			defer ms.Close(ctx)
+
+			req := motion.MoveOnMapReq{
+				ComponentName: base.Named("test-base"),
+				Destination:   goalInSLAMFrame,
+				SlamName:      slam.Named("test_slam"),
+				Extra:         extra,
+			}
+
+			executionID, err := ms.(*builtIn).MoveOnMapNew(context.Background(), req)
```

Comment:
Could we please give this context a timeout so that the test doesn't hang if we introduce a deadlock into MoveOnMapNew later?